core/iwasm/aot/aot_loader.c

```diff
@@ -123,6 +123,28 @@ GET_U64_FROM_ADDR(uint32 *addr)
     return u.val;
 }
 
+#if defined(BUILD_TARGET_XTENSA)
+
+static inline uint8
+GET_U8_FROM_ADDR(const uint8* p)
+{
+    bh_assert(p);
+    uint8 res = 0;
+
+    bh_memcpy_aw(&res, sizeof(uint8), p, sizeof(uint8));
+    return res;
+}
+
+static inline uint16
+GET_U16_FROM_ADDR(const uint8 *p)
+{
+    bh_assert(p);
+    uint16 res;
+
+    bh_memcpy_aw((uint8 *)&res, sizeof(uint16), p, sizeof(uint16));
```

Comment:
How about
```C
const uint8 *p_aligned = align_ptr(p + 3, 4) - 4;
uint32 buf32 = *(const uint32 *)p_aligned;
const uint8 *buf = (const uint8 *)&buf32;
if (is_little_endian())
    return (buf[p - p_aligned + 1] << 1) | buf[p - p_aligned];
else
    return (buf[p - p_aligned] << 1) | buf[p - p_aligned + 1];
```

Reply:
can it work also for string or array bytes?

Reply:
Not sure, maybe you can optimize the algorithm of bh_memcpy_aw:
copy the leading 1-3 unaligned bytes if there are
copy 4-byte aligned bytes
copy trailing 1-3 unaligned bytes if there are

Reply:
OK, I'd like the bh_memcpy_aw can work for any length target.

Reply:
> 
For "const uint8 *buf" , the intent is "const uint16 *buf"?  and the operator for shift lengh should be 8, not 1?
And shall we need judge is_little_endian? copy the bytes as they are in src will follow what it is.

Reply:
> For "const uint8 *buf" , the intent is "const uint16 *buf"? and the operator for shift lengh should be 8, not 1?  And shall we need judge is_little_endian? copy the bytes as they are in src will follow what it is.

Hi, I just give an example, you can implement it according to your system. And yes, my mistake, the shift length should be 8. Maybe you can:
```C
const uint8 *p_aligned = align_ptr(p + 3, 4) - 4;
uint32 buf32 = *(const uint32 *)p_aligned;
const uint8 *buf = (const uint8 *)&buf32;

res = *(uint16 *)(buf + p - p_aligned);
or
if (is_little_endian())
    return (buf[p - p_aligned + 8] << 8) | buf[p - p_aligned];
else
    return (buf[p - p_aligned] << 8) | buf[p - p_aligned + 8];
```